### PR TITLE
chore(arm): update identity output

### PR DESCRIPTION
**Pull request by xzf0587**
1. update identity output: identity resource id and identity client id
2. update function, bot, SQL reading the output
3. update unit test

### CODE REVIEW

File: packages/fx-core/src/plugins/resource/identity/config.ts [hunk 1]
```diff
@@ -7,6 +7,6 @@ export class IdentityConfig {
   location = "";
 
   identity = "";
```
Comment (dilin-MS) [inline packages/fx-core/src/plugins/resource/identity/config.ts:9]: What's this `identity` means?

File: packages/fx-core/src/plugins/resource/identity/config.ts [hunk 2]
```diff
@@ -7,6 +7,6 @@ export class IdentityConfig {
...
-  identityId = "";
-  identityName = "";
+  clientId = "";
```
Comment (dilin-MS) [inline packages/fx-core/src/plugins/resource/identity/config.ts:10]: what about `identityClientId`?

File: packages/fx-core/src/plugins/resource/identity/index.ts [hunk 3]
```diff
@@ -111,9 +111,9 @@ export class IdentityPlugin implements Plugin {
       return err(error);
     }
 
-    ctx.config.set(Constants.identityName, this.config.identityName);
-    ctx.config.set(Constants.identityId, this.config.identityId);
-    ctx.config.set(Constants.identity, this.config.identity);
+    ctx.config.set(Constants.identityName, this.config.identity);
+    ctx.config.set(Constants.identityClientId, this.config.clientId);
```
Comment (blackchoey) [inline packages/fx-core/src/plugins/resource/identity/index.ts:115]: Do we need to update the migration logic to change the name of related properties in `env.default.json`?

File: packages/fx-core/src/plugins/resource/identity/index.ts [hunk 4]
```diff
@@ -242,8 +242,14 @@ export class IdentityPlugin implements Plugin {
     const resourceId = getArmOutput(ctx, IdentityArmOutput.identityName);
 
     ctx.config.set(Constants.identityName, resourceId);
```
Comment (blackchoey) [inline packages/fx-core/src/plugins/resource/identity/index.ts:244]: do we also need to update this line? the identity name is no longer resource id

File: packages/fx-core/src/plugins/resource/function/ops/provision.ts [hunk 5]
```diff
@@ -179,7 +179,7 @@ export class FunctionProvision {
     site.identity = {
       type: DefaultProvisionConfigs.siteIdentityTypeUserAssigned,
       userAssignedIdentities: {
-        [identityName]: {},
+        [identityResourceId]: {},
```
Comment (hund030) [inline packages/fx-core/src/plugins/resource/function/ops/provision.ts:182]: We simply replace identity name with resource id, does this also work?

**Review by YitongFeng-git — APPROVED** [review]

File: packages/fx-core/src/plugins/resource/function/ops/provision.ts (hunk 5 above)
Comment (xzf0587) [reply packages/fx-core/src/plugins/resource/function/ops/provision.ts:182]: We change the key name, but the value is always the resource id.

**Review by blackchoey — APPROVED** [review]

**Review by YitongFeng-git — APPROVED** [review]

**Pull request merged**